### PR TITLE
feat: Budget category alerts settings UI

### PR DESCRIPTION
**Pull request by ptbrowne**
- 🖱 UI is functional : a new section in the settings allows to configure
an budget alert for a category. Alerts are shown as clickable cards that
open a modal to configure the alert.

- ✏️ Saves alerts into the io.cozy.bank.settings document under the `categoryBudgetAlerts` key

- 🚩 Section inside settings only shown if `settings.budget-category-alerts` flag is set.

Alert schema:

```
{
  accountOrGroup: {
    _id: 'c0ffeedeadbeef',
    _type: 'io.cozy.bank.accounts'
  }, // can be null if the alert concerns all accounts
  balanceMinThresold: 100,
  category: '400100'
}
```

### CODE REVIEW

File: src/ducks/settings/CategoryAlerts/CategoryAlertCard.jsx [hunk 1]
```diff
@@ -0,0 +1,121 @@
...
+        />
+      ) : null}
+      {confirmingRemoval ? (
+        <Modal
+          primaryText={t('Settings.budget-category-alerts.remove.ok')}
+          primaryType="danger"
+          primaryAction={handleRemoveAlert}
+          secondaryText={t('Settings.budget-category-alerts.remove.cancel')}
+          secondaryAction={() => setConfirmingRemoval(false)}
+          dismissAction={() => setConfirmingRemoval(false)}
```
Comment (drazik) [inline src/ducks/settings/CategoryAlerts/CategoryAlertCard.jsx]: Nit: extract `setConfirmingRemoval(false)` to a function like `hideRemovalConfirm`

File: src/ducks/settings/CategoryAlerts/CategoryAlertSettingsPane.jsx [hunk 2]
```diff
@@ -0,0 +1,158 @@
...
+    : [...arr, item]
+}
+
+const maxValue = (arr, fn) => {
+  const rs = arr.reduce((curMax, item) => {
+    const v = fn(item)
+    return typeof v === 'number' ? Math.max(v, curMax) : curMax
+  }, 0)
+  return rs
+}
```
Comment (drazik) [inline src/ducks/settings/CategoryAlerts/CategoryAlertSettingsPane.jsx]: Isn't Lodash's maxBy doing the same job?

File: src/ducks/settings/CategoryAlerts/CategoryAlertSettingsPane.jsx [hunk 3]
```diff
@@ -0,0 +1,158 @@
...
+      <TogglePaneTitle>
+        {t('Settings.budget-category-alerts.pane-title')}
+      </TogglePaneTitle>
+      <TogglePaneText>
+        {t('Settings.budget-category-alerts.pane-description')}
+      </TogglePaneText>
+      <Stack spacing="s">
+        <div>
+          <Grid>
+            {!alerts
```
Comment (drazik) [inline src/ducks/settings/CategoryAlerts/CategoryAlertSettingsPane.jsx]: Nit: When possible, I prefer positive conditions

File: src/ducks/settings/CategoryAlerts/CategoryAlertCard.jsx (hunk 1 above)
Comment (ptbrowne) [reply src/ducks/settings/CategoryAlerts/CategoryAlertCard.jsx]: Why ? Do you think the name is not clear enough ?

File: src/ducks/settings/CategoryAlerts/CategoryAlertSettingsPane.jsx (hunk 2 above)
Comment (ptbrowne) [reply src/ducks/settings/CategoryAlerts/CategoryAlertSettingsPane.jsx]: maxBy will return the item, not the value. But I agree, it is simpler to use maxBy, I changed the code to use it.

**Pull request merged**